resolver/src/main/java/io/netty/resolver/DefaultHostsFileEntriesResolver.java

```diff
@@ -17,49 +17,46 @@
 
 import io.netty.util.CharsetUtil;
 import io.netty.util.internal.PlatformDependent;
+import io.netty.util.internal.SystemPropertyUtil;
 
 import java.net.InetAddress;
 import java.nio.charset.Charset;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Locale;
 import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
 
 /**
  * Default {@link HostsFileEntriesResolver} that resolves hosts file entries only once.
  */
 public final class DefaultHostsFileEntriesResolver implements HostsFileEntriesResolver {
+    private static final long DEFAULT_REFRESH_INTERVAL = TimeUnit.SECONDS.toNanos(60);
 
-    private final Map<String, List<InetAddress>> inet4Entries;
-    private final Map<String, List<InetAddress>> inet6Entries;
+    private final long refreshInterval;
+    private final AtomicLong lastRefresh = new AtomicLong(System.nanoTime());
+    private final HostsFileEntriesProvider.Parser hostsFileParser;
+    private volatile Map<String, List<InetAddress>> inet4Entries;
+    private volatile Map<String, List<InetAddress>> inet6Entries;
 
     public DefaultHostsFileEntriesResolver() {
-        this(parseEntries());
+        this(HostsFileEntriesProvider.parser());
     }
 
     // for testing purpose only
-    DefaultHostsFileEntriesResolver(HostsFileEntriesProvider entries) {
+    DefaultHostsFileEntriesResolver(HostsFileEntriesProvider.Parser hostsFileParser) {
+        this.hostsFileParser = hostsFileParser;
+        this.refreshInterval = SystemPropertyUtil.getLong(
+                "io.netty.hostsFileRefreshInterval", DEFAULT_REFRESH_INTERVAL);
```

Comment:
just do this in a init block like we do in other places:

https://github.com/netty/netty/blob/netty-4.1.72.Final/buffer/src/main/java/io/netty/buffer/PooledByteBufAllocator.java#L117